server/datastore/mysql/software_test.go

```diff
@@ -5975,6 +6054,26 @@ func testListHostSoftwareWithLabelScoping(t *testing.T, ds *Datastore) {
 	require.NoError(t, err)
 	checkSoftware(software)
 
+	software, _, err = ds.ListHostSoftware(ctx, anotherHost, opts)
+	require.NoError(t, err)
+	require.Len(t, software, 2)
+	require.Equal(t, software[0].Name, softwareAlreadyInstalled.Name)
+	require.Equal(t, software[0].InstalledVersions[0].Version, softwareAlreadyInstalled.Version)
+	require.Equal(t, software[0].SoftwarePackage.Version, installer1.Version)
+	require.Equal(t, software[1].Name, mutationResults.Inserted[0].Name)
+	require.Nil(t, software[1].InstalledVersions)
```

Comment:
This seems like it should be a nonzero-length list, given the SQL executed starting ~L5994? Since `anotherHost` has the software installed?

Reply:
Actually, we're good here. I got turned around. Added more comments to explain what's going on here in the next commit.